packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -3,50 +3,169 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:ui';
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/services.dart';
 
-/// A leaf node in the focus tree that can receive focus.
+import 'binding.dart';
+import 'framework.dart';
+
+typedef FocusNodeVisitor = bool Function(FocusNode node);
+
+typedef FocusOnKeyCallback = bool Function(FocusNode node, RawKeyEvent event);
+
+
+/// An object that can be used by a stateful widget to obtain the keyboard focus
+/// and to handle keyboard events.
+///
+/// {@template flutter.widgets.focus_manager.focus.advancedWarning}
```

Comment:
Hmm, yeah.  I added a clause about "or keeping [FocusNode] references so that you can control the focus from another part of the application", which hopefully addresses that.